From: =?UTF-8?q?Mika=20Pfl=C3=BCger?= <debian@mikapflueger.de>
Date: Sun, 4 Mar 2012 00:25:43 +0100
Subject: Webalizer policy adjustments: * Labeled awffull as webalizer_exec_t
 * Allow webalizer to read usr_t files for geoip database * Allow
 logrotate_t to transition to webalizer_t for web log processing

---
 policy/modules/admin/logrotate.te |    4 ++++
 policy/modules/apps/webalizer.fc  |    1 +
 policy/modules/apps/webalizer.te  |    7 ++-----
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/policy/modules/admin/logrotate.te b/policy/modules/admin/logrotate.te
index 7090dae..a8f15b2 100644
--- a/policy/modules/admin/logrotate.te
+++ b/policy/modules/admin/logrotate.te
@@ -162,6 +162,10 @@ optional_policy(`
 ')
 
 optional_policy(`
+	webalizer_domtrans(logrotate_t)
+')
+
+optional_policy(`
 	consoletype_exec(logrotate_t)
 ')
 
diff --git a/policy/modules/apps/webalizer.fc b/policy/modules/apps/webalizer.fc
index e4f7d30..989d69c 100644
--- a/policy/modules/apps/webalizer.fc
+++ b/policy/modules/apps/webalizer.fc
@@ -3,6 +3,7 @@
 # /usr
 #
 /usr/bin/webalizer	--	gen_context(system_u:object_r:webalizer_exec_t,s0)
+/usr/bin/awffull	--	gen_context(system_u:object_r:webalizer_exec_t,s0)
 
 #
 # /var
diff --git a/policy/modules/apps/webalizer.te b/policy/modules/apps/webalizer.te
index 75629b6..c42c93e 100644
--- a/policy/modules/apps/webalizer.te
+++ b/policy/modules/apps/webalizer.te
@@ -13,17 +13,13 @@ role system_r types webalizer_t;
 type webalizer_etc_t;
 files_config_file(webalizer_etc_t)
 
-type webalizer_usage_t;
-files_type(webalizer_usage_t)
-
 type webalizer_tmp_t;
 files_tmp_file(webalizer_tmp_t)
 
 type webalizer_var_lib_t;
 files_type(webalizer_var_lib_t)
 
-type webalizer_write_t;
-files_type(webalizer_write_t)
+typealias webalizer_var_lib_t alias { webalizer_write_t webalizer_usage_t };
 
 ########################################
 #
@@ -71,6 +67,7 @@ fs_rw_anon_inodefs_files(webalizer_t)
 
 files_read_etc_files(webalizer_t)
 files_read_etc_runtime_files(webalizer_t)
+files_read_usr_files(webalizer_t)
 
 logging_list_logs(webalizer_t)
 logging_send_syslog_msg(webalizer_t)
